### PR TITLE
Replace reindex-aip-data with more generic reindex-index-data

### PR DESCRIPTION
**Pull request by Hwesta**
Rename reindex-aip-data to reindex-index data and update for transfers, move doc_type processing into loops. Update README.

Feedback on README text encouraged.

### CODE REVIEW

**Comment by Hwesta** [comment]
Potential improvement is to have the script stop and restart if interrupted.  This could be done by saving out the index contents to a json file before it deletes them, and have the ability to read those files back in in a subsequent run.

File: README.md [hunk 1]
```diff
@@ -85,3 +85,13 @@ Edges are labelled with the user choice or exit code that connects those nodes,
...
   * Destination: `Transfer.magicLink`. This is set by the most recent 'assign magic link' (`3590f73d-5eb0-44a0-91a6-5b2db6655889`)
+
+### reindex-index-data
+
+reindex-index-data will delete and re-index an ElasticSearch index, creating it with an updated mapping based on the currently installed Archivematica instance.
+This is useful if the mapping has changed in an incompatible way but the existing data should be preserved.
+Depending on the size of the index, a large amount of memory may be consumed.
+
+There is one parameter which specifies which index to recreate: `transfers` or `aips`.
+The script currently requires `DJANGO_SETTINGS_MODULE` in the environment.
```
Comment (mistydemeo) [inline README.md]: Let's get #17 in, and then it should be unnecesary to document this!

File: tools/reindex-index-data [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        for r in records:
+            conn.index(body=r, index=index, doc_type=doc_type)
+        print('Done indexing', len(records), index + '/' + doc_type, 'records.')
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Regenerate index based on the existing data to update the index types.')
+    parser.add_argument('index', choices=['transfers', 'aips'], help='Index to re-generate.')
+    args = parser.parse_args()
+
+    if args.index == 'transfers':
```
Comment (mistydemeo) [inline tools/reindex-index-data]: How about storing these in a dict instead of using an if statement?

**Comment by mistydemeo** [comment]
:+1:

**Pull request merged**